components/raftstore/src/store/snap.rs

```diff
@@ -1966,6 +1990,66 @@ impl SnapManagerCore {
     }
 }
 
+/// `SnapRecvConcurrencyLimiter` enforces a limit on the number of simultaneous
+/// snapshot receives. It is consulted before a snapshot is generated.
+/// employs a TTL mechanism to automatically evict operations that have been
+/// pending longer than the specified TTL. The TTL helps to handle scenarios
+/// where a snapshot fails to be sent for any reason.
+#[derive(Clone)]
+pub struct SnapRecvConcurrencyLimiter {
+    limit: Arc<AtomicUsize>,
+    ttl_secs: u64,
+    timestamps: Arc<Mutex<VecDeque<Instant>>>,
```

Comment:
We should shrink its capacity periodically.

Reply:
This queue already has an upper bound on its size, which is `limit`. Since `limit` is usually very small (default is 32), do we need to be concerned with its memory usage? Also, we would expire stale entries in the queue whenever the follower requests a snapshot, which I assume is a fairly frequent event?

Reply:
I see, it's better to initialize the vecdeque using `with_capacity()`.

Reply:
sure, I can add that in